### PR TITLE
Update operations/source model for receptor-enabled availability checks

### PR DESCRIPTION
**Pull request by lindgrenj6**
Even if this doesn't get used, these are the changes that I needed to make in order for availability checks with a receptor_client. 

\# TODO:
- [x] ~specs~ taken care of in sub-classes since it is all in `Source#connection_check`

cc @syncrou @slemrmartin

### CODE REVIEW

File: lib/topological_inventory/providers/common/operations/source.rb [hunk 1]
```diff
@@ -121,7 +124,11 @@ def endpoint
         end
 
         def authentication
-          @authentication ||= api_client.fetch_authentication(source_id, endpoint)
+          @authentication ||= if endpoint.receptor_node.present?
+                                "n/a"
```
Comment (slemrmartin) [inline lib/topological_inventory/providers/common/operations/source.rb]: I suppose it'll be tested somewhere against this value so I would prefer constant here

Comment (lindgrenj6) [reply lib/topological_inventory/providers/common/operations/source.rb]: done!

File: lib/topological_inventory/providers/common/operations/source.rb [hunk 2]
```diff
@@ -17,13 +17,17 @@ class Source
         }.freeze
 
         LAST_CHECKED_AT_THRESHOLD = 5.minutes.freeze
+        AUTH_NOT_NECESSARY = "n/a".freeze
 
-        attr_accessor :params, :request_context, :source_id
+        attr_accessor :params, :request_context, :source_id, :account_number, :receptor_client
 
-        def initialize(params = {}, request_context = nil)
+        def initialize(params = {}, request_context = nil, receptor_client = nil)
```
Comment (slemrmartin) [inline lib/topological_inventory/providers/common/operations/source.rb]: this change is not needed now (regarding https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/107)

Comment (lindgrenj6) [reply lib/topological_inventory/providers/common/operations/source.rb]: Updated!

**Review by slemrmartin — APPROVED** [review]
:+1: LGTM

**Pull request merged**